eng/pipelines/templates/steps/split-tests.yml

```diff
@@ -0,0 +1,55 @@
+parameters:
+- name: OutputFolder
+  type: string
+  default: "projects"
+- name: numOfTestsPerJob
+  type: number
+  default: 20
+- name: OutputProjectFilePath
+  type: string
+  default: "$(System.DefaultWorkingDirectory)/projects.txt"
+- name: TestTargetDependency
+  type: string
+  default: "not-specific"
+- name: ProjectListOverrideFile
+  type: string
+  default: "ProjectListOverrideFile"
+steps:
+  - pwsh: |
+      dotnet build /t:ProjectDependsOn ./eng/service.proj `
+        /p:ServiceDirectory=* /p:TestTargetDependency="${{parameters.TestTargetDependency}}" `
+        /p:IncludeSrc=false /p:IncludeStress=false /p:IncludeSamples=false  `
+        /p:IncludePerf=false /p:RunApiCompat=false `
+        /p:InheritDocEnabled=false /p:BuildProjectReferences=false `
+        /p:BuildInParallel=true /p:OutputProjectFilePath="${{parameters.OutputProjectFilePath}}"
+    displayName: Find all test dependencies
+
+  - task: PowerShell@2
+    inputs:
+      targetType: filePath
+      filePath: eng/scripts/Split-Tests-Xml.ps1
+      arguments: >-
+        -ProjectFilePath ${{ parameters.OutputProjectFilePath }}
+        -OutputFolder ${{ parameters.OutputFolder }}
+        -numOfTestsPerJob ${{ parameters.numOfTestsPerJob }}
+      ignoreLASTEXITCODE: true
+      pwsh: true
+    displayName: Split testing packages into different files
+
+  - pwsh: |
+      $platformJson = Get-Content "./eng/pipelines/templates/stages/platform-matrix.json" | ConvertFrom-Json
```

Comment:
We should not hard code the matrix we should pass through the MatrixConfigs parameter and maybe even consider an allowing for passing through a specific matrix config for the dependency tests.